gdal: update to 3.6.4

nilason opened the pull request:
#### Description

<!-- Note: it is best to make pull requests from a branch rather than from master -->

Updates GDAL to version 3.6.3.

GDAL has dropped support for autotools, thus this is a complete remake to cmake. 

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->
macOS 12.6.3 21G419 arm64
Xcode 14.2 14C18

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

nilason commented:
Just noticed that there are two Python packages installed outside MacPorts directory in the CI build log:

```
/Library/Python/3.9/site-packages/osgeo/
/Library/Python/3.9/site-packages/osgeo_utils/
```

This didn't happen locally (it went to `/opt/local/Library/Frameworks/Python.framework/Versions/3.11/...`)

It is possible to configure which Python to use:
see [GDAL docs](https://gdal.org/development/building_from_source.html#python-bindings-options) for options.

I'm not sure what would be the correct or preferable way to detect and set specific Python version. I'd be grateful for any help on this!

nilason commented:
> Just noticed that there are two Python packages installed outside MacPorts directory in the CI build log:
> 
> ```
> /Library/Python/3.9/site-packages/osgeo/
> /Library/Python/3.9/site-packages/osgeo_utils/
> ```
> 
> This didn't happen locally (it went to `/opt/local/Library/Frameworks/Python.framework/Versions/3.11/...`)
> 
> It is possible to configure which Python to use: see [GDAL docs](https://gdal.org/development/building_from_source.html#python-bindings-options) for options.
> 
> I'm not sure what would be the correct or preferable way to detect and set specific Python version. I'd be grateful for any help on this!

How silly I am, this is what `py-gdal` is for.

ryandesign commented on gis/gdal/Portfile:
```diff
@@ -505,85 +454,90 @@ foreach s ${postgresql_suffixes} {
     depends_lib-append      port:cfitsio
-    configure.args-delete   --without-cfitsio
-    configure.args-append   --with-cfitsio
+    configure.args-delete   -DGDAL_USE_CFITSIO=OFF
+    configure.args-append   -DGDAL_USE_CFITSIO=ON
```
```suggestion
    configure.args-replace  -DGDAL_USE_CFITSIO=OFF  -DGDAL_USE_CFITSIO=ON
```

ryandesign commented on gis/gdal/Portfile:
```diff
@@ -505,85 +454,90 @@ foreach s ${postgresql_suffixes} {
+        variant proj${proj_ver} description "Use Proj${proj_ver}" conflicts {*}${cflcts} "
+
+            depends_lib-append      port:proj${proj_ver}
+            configure.args-append   -DGDAL_USE_PROJ=ON
+            configure.args-append   -DPROJ_INCLUDE_DIR=${prefix}/lib/proj${proj_ver}/include
+            configure.args-append   -DPROJ_LIBRARY_RELEASE=${prefix}/lib/proj${proj_ver}/lib/libproj.dylib
```
```suggestion
            configure.args-append   -DGDAL_USE_PROJ=ON \
                                    -DPROJ_INCLUDE_DIR=${prefix}/lib/proj${proj_ver}/include \
                                    -DPROJ_LIBRARY_RELEASE=${prefix}/lib/proj${proj_ver}/lib/libproj.dylib
```

ryandesign commented on gis/gdal/Portfile:
```diff
@@ -67,214 +67,202 @@ depends_lib-append  port:zlib \
+                    -DGDAL_ENABLE_DRIVER_USGSDEM=ON          \
+                    -DGDAL_ENABLE_DRIVER_XPM=ON              \
+                    -DGDAL_ENABLE_DRIVER_XYZ=ON              \
+                    -DGDAL_ENABLE_DRIVER_ZMAP=ON
```
Are you sure all these flags are correct? The CI logs say:
```
CMake Warning:
    Manually-specified variables were not used by the project:
  
      CMAKE_OBJCXX_COMPILER
      CMAKE_OBJC_COMPILER
      CMAKE_POLICY_DEFAULT_CMP0025
      CMAKE_POLICY_DEFAULT_CMP0060
      GDAL_ENABLE_DRIVER_AIG
      GDAL_ENABLE_DRIVER_ESAT
      GDAL_ENABLE_DRIVER_NOAA_B
      GDAL_ENABLE_DRIVER_NSIDCBIN
      GDAL_ENABLE_DRIVER_OPENFILEGDB
      GDAL_ENABLE_DRIVER_WFS
      GDAL_USE_CHARLS
      GDAL_USE_CRYPTO
      GDAL_USE_EXR
      GDAL_USE_FME
      GDAL_USE_JASPER
      GDAL_USE_JP2MRSID
      GDAL_USE_KAKADU
      GDAL_USE_LIBZ
      GDAL_USE_MONGOCXXV3
      GDAL_USE_MSG
      GDAL_USE_OCI
      GDAL_USE_PAM
      GDAL_USE_PERL
      GDAL_USE_PG
      GDAL_USE_PYTHON
      GDAL_USE_XERCES
      GDAL_USE_XML2
      OGR_ENABLE_DRIVER_AVCBIN
      OGR_ENABLE_DRIVER_AVCE00
      OGR_ENABLE_DRIVER_GTFS
      OGR_ENABLE_DRIVER_ZARR
```

nilason replied:
> Are you sure all these flags are correct? The CI logs say:
> 
> ```
> CMake Warning:
>     Manually-specified variables were not used by the project:
>   
>       CMAKE_OBJCXX_COMPILER
> ...
> ```

These were all addressed, either removed or renamed.

ryandesign reviewed (commented):
> Just noticed that there are two Python packages installed outside MacPorts directory in the CI build log:
> 
> ```
> /Library/Python/3.9/site-packages/osgeo/
> /Library/Python/3.9/site-packages/osgeo_utils/
> ```

You'll need to figure out how to get the build system to stop doing that.

nilason commented on gis/gdal-grass/Portfile:
```diff
@@ -1,65 +1,86 @@
+checksums           md5     1ee8603aa10a33fcaf5939e587d19248 \
+                    rmd160  13e754bf956c42649f3e82e1a81bdbf6165c2940 \
+                    sha256  f703edc94c0177be2627219f0abc0fb02635c5e589af730240c2ec482be76acf \
+                    size    58068
```
> f this project offers a manually-created tarball, please use that. If not, please specify github.tarball_from archive.

At this time there is no manually created tarball, I changed to `github.tarball_from archive`.

nilason commented on gis/gdal-grass/Portfile:
```diff
@@ -1,65 +1,86 @@
+revision            0
+epoch               1
+
+github.setup        OSGeo gdal-grass 1.0.2
```
This is the reason I set the `epoch`, I really hope that will be enough, as I was at least partially responsible for this decision (by not objecting).

Previously this port was part of GDAL and followed its versioning, but was separated because of circular dependency issues with GRASS. It is now a separate "addon" driver to GDAL. The idea with resetting the numbering, was to start fresh...

ryandesign replied:
> This is the reason I set the `epoch`, I really hope that will be enough

It will be enough as long as we also remember never to publish 1.4.3_0, 2.1.0_0, 2.1.0_1, 2.2.0_0, 2.2.0_1, or 2.2.0_2 again in this new version numbering system.

nilason replied:
> > This is the reason I set the `epoch`, I really hope that will be enough
> 
> It will be enough as long as we also remember never to publish 1.4.3_0, 2.1.0_0, 2.1.0_1, 2.2.0_0, 2.2.0_1, or 2.2.0_2 again in this new version numbering system.

Added this info in a reminder comment in Portfile. Thanks!

nilason commented:
@ryandesign Thanks a lot for taking time on this! I think I addressed all you suggestions, issues.
I added a minor liblas update which I may remove if not appropriate in one PR.

nilason commented:
> > Just noticed that there are two Python packages installed outside MacPorts directory in the CI build log:
> > ```
> > /Library/Python/3.9/site-packages/osgeo/
> > /Library/Python/3.9/site-packages/osgeo_utils/
> > ```
> 
> You'll need to figure out how to get the build system to stop doing that.

Setting `-DBUILD_PYTHON_BINDINGS` was needed to drop them.

nilason commented:
@ryandesign What is the common practice here, shall I add a commit with revision bump of dependants?

nilason commented:
@Veence Please take a look at this. GDAL update is long overdue.

nilason commented on gis/gdal-grass/Portfile:
```diff
@@ -1,65 +1,84 @@
+pre-configure {
+    set grass_gisbase [exec ${prefix}/bin/grass --config path]
 
-configure.args-append   --with-gdal=${prefix}/bin/gdal-config
-configure.args-append   --with-grass=${prefix}/share/grass-7.4.0
+    configure.args-append --with-grass=${grass_gisbase}
```
Fixes https://trac.macports.org/ticket/64411

nilason commented on gis/gdal/Portfile:
```diff
@@ -505,85 +445,84 @@ foreach s ${postgresql_suffixes} {
     "
 }
 
-variant mysql5 conflicts mysql57 mysql8 description {Enable MySQL 5 support} {
```
Addresses https://trac.macports.org/ticket/43431

reneeotten commented on gis/gdal/Portfile:
```diff
@@ -292,69 +275,48 @@ pre-configure {
+variant libkml description {Enable libkml} {
+    PortGroup               boost 1.0
 
-    configure.args-delete   --without-grass
-    configure.args-append   --with-grass=${prefix}/share/grass78
+    depends_lib-append      port:libkml
+    boost.depends_type      build
```
not sure exactly what is needed here, but the `boost` PG does set quite a few `cmake` related variables already (see [here](https://github.com/macports/macports-ports/blob/7d3cdee471ccc3514bea2eb1686c975249578a06/_resources/port1.0/group/boost-1.0.tcl#L147)). If the `gdal` port needs another variable name the PG probably will provide the necessary hints to do so.
Also, since you're setting the dependency type to `build` it would be good to check that nothing links to `boost` package (i.e., is required at runtime).

nilason replied:
It is not GDAL that uses boost, but libkml. The GDAL driver code includes libkml header files which in turn includes boost. GDAL expects the header files to be in standard `${prefix}/include` and has no way to know to look into eg.  `${prefix}/libexec/boost/1.76/include`.

nilason replied:
> Also, since you're setting the dependency type to `build` it would be good to check that nothing links to `boost` package (i.e., is required at runtime).


GDAL doesn't seem to link (as its not using boost directly):
```
otool -L /opt/local/lib/gdalplugins/ogr_LIBKML.dylib
/opt/local/lib/gdalplugins/ogr_LIBKML.dylib:
	@rpath/libgdal.32.dylib (compatibility version 32.0.0, current version 32.3.6)
	/opt/local/lib/libkmlbase.1.dylib (compatibility version 1.0.0, current version 1.3.0)
	/opt/local/lib/libkmldom.1.dylib (compatibility version 1.0.0, current version 1.3.0)
	/opt/local/lib/libkmlengine.1.dylib (compatibility version 1.0.0, current version 1.3.0)
	/opt/local/lib/libminizip.1.dylib (compatibility version 2.0.0, current version 2.0.0)
	/opt/local/lib/liburiparser.1.dylib (compatibility version 1.0.0, current version 1.0.30)
	/opt/local/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.13)
	/opt/local/lib/libexpat.1.dylib (compatibility version 10.0.0, current version 10.10.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
	/opt/local/lib/openmpi-mp/libmpi_cxx.40.dylib (compatibility version 71.0.0, current version 71.1.0)
	/opt/local/lib/openmpi-mp/libmpi.40.dylib (compatibility version 71.0.0, current version 71.4.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1300.23.0)
```

reneeotten replied:
> It is not GDAL that uses boost, but libkml. The GDAL driver code includes libkml header files which in turn includes boost. GDAL expects the header files to be in standard `${prefix}/include` and has no way to know to look into eg. `${prefix}/libexec/boost/1.76/include`.

that sounds strange to me... if `gdal` doesn't use `boost` then it shouldn't need to have it as a built dependency. If the `libkml` port uses `boost` then it should probably specify in that port where the boost headers are and `gdal` should be able to use those - or am I missing something here?

nilason replied:
> that sounds strange to me... if `gdal` doesn't use `boost` then it shouldn't need to have it as a built dependency. 

There's no need for `GDAL` to have `boost` as dependency. With the GDAL variant (and port) `libkml`, which *does* have `boost` as dependency, however, GDAL's libkml-driver hooks to the libkml API (which in turn does, correctly, include boost API). libkml know where to look for boost header files, but GDAL doesn't. This would be all good and fine if they were located in "standard" location for include files (i.e. `${prefix}/include`) .

> If the `libkml` port uses `boost` then it should probably specify in that port where the boost headers are and `gdal` should be able to use those - or am I missing something here?

The boost PG defines the version used and ultimately the location of the include files. This is the reason, the only solution I could come up with, was the present by adding the boost PG to GDAL variant libkml. Note this only affects GDAL with the libkml variant. If there is a way to get the include path to boost, from boost PG, without adding the PG to GDAL, that would be an alternative and perhaps better solution.

nilason replied:
In short: if it is possible to get:
https://github.com/macports/macports-ports/blob/7d3cdee471ccc3514bea2eb1686c975249578a06/_resources/port1.0/group/boost-1.0.tcl#L52
without adding boost PG to gdal+libkml, then that would likely be enough.

ryandesign commented:
When I installed on my system in trace mode (`sudo port -t install`), it reported a lot of files provided by ports that were not declared as dependencies that the build would have accessed if I had not used trace mode:

```
  /opt/local/include/H5Cpp.h
  /opt/local/include/OpenEXR/ImfVersion.h
  /opt/local/include/bsoncxx/v_noabi/bsoncxx/config/version.hpp
  /opt/local/include/hdf5.h
  /opt/local/include/json-c/json.h
  /opt/local/include/libheif/heif.h
  /opt/local/include/mongocxx/v_noabi/mongocxx/client.hpp
  /opt/local/include/msodbcsql.h
  /opt/local/include/openjpeg-2.5/opj_config.h
  /opt/local/include/pcre.h
  /opt/local/include/pcre2.h
  /opt/local/include/sql.h
  /opt/local/lib/cmake/json-c/json-c-config.cmake
  /opt/local/lib/oracle/sdk/include/oci.h
  /opt/local/lib/pkgconfig/OpenEXR.pc
  /opt/local/lib/pkgconfig/json-c.pc
  /opt/local/lib/pkgconfig/libheif.pc
  /opt/local/lib/pkgconfig/libmongocxx.pc
  /opt/local/lib/pkgconfig/libopenjp2.pc
```

Often this indicates that opportunistic linking will occur, which we don't want, and if so the build should be changed so that it doesn't access those files unless it declares dependencies on those ports.

gdal 3.4.3 installed libgdal.30.dylib while gdal 3.6.3 installs libgdal.32.dylib so all ports that link with the library will be broken by this update and need to be rebuilt (via a version update or a revision increase) to avoid that. Here's the list of all ports that mention a gdal dependency though I don't know if all of them link with the library:

```
% port file all|sort -u|xargs grep -i ':gdal\b'
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/databases/postgis2/Portfile:    depends_lib-append      port:gdal
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/databases/postgis3/Portfile:    depends_lib-append      port:gdal
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/gdal-grass/Portfile:depends_lib         port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/grass/Portfile:                    port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/grass7/Portfile:                    port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/kealib/Portfile:depends_lib-append  port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/liblas/Portfile:depends_lib-append  port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/libosmium/Portfile:                        port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/mapnik/Portfile:    depends_lib-append      port:gdal
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/mapserver/Portfile:    depends_lib-append      port:gdal
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/mapserver/Portfile:    depends_lib-append      port:gdal
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/orfeotoolbox/Portfile:                    port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/pdal/Portfile:                    port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/qgis3/Portfile:                    port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/qlandkartegt/Portfile:depends_lib      port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/rsgislib/Portfile:depends_lib-append  port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/saga/Portfile:                    port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/gis/t-rex/Portfile:    port:gdal
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/graphics/OpenSceneGraph-devel/Portfile:                        port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/graphics/OpenSceneGraph/Portfile:                        port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/graphics/opencv3-devel/Portfile:                            port:gdal
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/graphics/opencv3/Portfile:                            port:gdal
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/graphics/opencv4-devel/Portfile:                    port:gdal
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/graphics/opencv4/Portfile:                    port:gdal
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/octave/octave-mapping/Portfile:depends_lib-append  port:gdal
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/python/py-fiona/Portfile:                        port:gdal
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/python/py-gdal/Portfile:                    port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/python/py-rasterio/Portfile:                        port:gdal
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/science/gerbil/Portfile:                    port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/science/gmt4/Portfile:    depends_lib-append      port:gdal
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/science/gmt5/Portfile:    depends_lib-append      port:gdal
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/science/ncarg/Portfile:                            port:gdal       \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/science/sumo/Portfile:    port:gdal \
/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/science/xastir/Portfile:                    port:gdal \
```

reneeotten commented:
thanks @ryandesign for pointing that out! Unfortunately trace-mode is still broken on newer OSes so it's hard to test this locally for  many people. There are also several Trac tickets for this port that I don't know if they have been resolved. Again, that's really the job of the maintainer of this port. 

This PR updates the port and doesn't break it more than it is in the current state. 
So yes, ideally everything would be fixed before merging - but at some point we probably just have to merge it and fix other issues step-by-step.

Dave-Allured commented:
Please update this PR from GDAL 3.6.3 to 3.6.4, a bug fix release on 2023 Apr 21.  Thanks.

nilason commented:
> thanks @ryandesign for pointing that out! Unfortunately trace-mode is still broken on newer OSes so it's hard to test this locally for many people. There are also several Trac tickets for this port that I don't know if they have been resolved. Again, that's really the job of the maintainer of this port.

@ryandesign Thanks again for looking at this! Even though I regularly use `port -vst` I have not seen something similar to what you listed. I do check what drivers are enabled and not, and whether the related port is enabled or not. There are sooo many of them so I cannot guarantee there aren't mismatches, but I general it should be ok. I guess cmake does check up on them (causing the trace), but if not explicitly told to be included they will be disabled.
 
> This PR updates the port and doesn't break it more than it is in the current state. So yes, ideally everything would be fixed before merging - but at some point we probably just have to merge it and fix other issues step-by-step.

I agree. I will make the bump to 3.6.4 (thanks @Dave-Allured!), and when there is an agreement on the boost issue, I'll add a dependencies-revision-bump-commit.

Dave-Allured commented:
@ryandesign said:
> When I installed on my system in trace mode (`sudo port -t install`), it reported a lot of files provided by ports that were not declared as dependencies that the build would have accessed if I had not used trace mode:
>
> ... Often this indicates that opportunistic linking will occur, which we don't want, and if so the build should be changed so that it doesn't access those files unless it declares dependencies on those ports.

Ryan, this seems like false positives.  I tested this with a default Macports build of GDAL 3.6.3, with custom prefix and no sudo privs.  It built fine, other dependencies were installed, there were no complaints, and none of those 19 reported files were installed by anything.  By name I associated 9 files out of 19 with inactive variants in @nilason's port file:
```
/opt/local/include/H5Cpp.h
/opt/local/include/hdf5.h
-DGDAL_USE_HDF5=OFF

/opt/local/include/OpenEXR/ImfVersion.h
/opt/local/lib/pkgconfig/OpenEXR.pc
-DGDAL_USE_OPENEXR=OFF

/opt/local/include/libheif/heif.h
/opt/local/lib/pkgconfig/libheif.pc
-DGDAL_USE_HEIF=OFF

/opt/local/include/openjpeg-2.5/opj_config.h
/opt/local/lib/pkgconfig/libopenjp2.pc
-DGDAL_USE_OPENJPEG=OFF

/opt/local/include/sql.h
-DGDAL_USE_MYSQL=OFF
```
So what is `port install -t` really doing here?  Testing all variants, or what?  Unless I am missing something, it seems that no new dependency declarations are needed for this port.

nilason commented:
Updated PR to bump to GDAL 3.6.4. Still not sure how to solve the boost issue differently.

nilason commented:
The commits https://github.com/macports/macports-ports/pull/18077/commits/41b2884b80c4c81f857b548bba43e3583ec4a443 and https://github.com/macports/macports-ports/pull/18077/commits/f32e5c89976b4e389dfe7bf40ce188bb14205889 is an alternative to initial PG solution for boost.

nilason commented:
I believe this is ready for merge.
My solution for the boost problem was a minor patch to the libkml pc-file.

nilason commented:
#18418 should preferably be merged before this.

nilason commented:
If there are no more objections, suggestions, this may be merged now.
More than a whole year since last version bump of this port, now being __very__ outdated, and a month since I put up this PR...

i0ntempest commented:
Are those failing checks just timeouts? If so I'll merge in 24 hours if no one objects.

reneeotten commented:
> Are those failing checks just timeouts? If so I'll merge in 24 hours if no one objects.

You haven't commented or reviewed anything on this PR; so I'd suggest you leave merging of this to the people who did and can verify whether the comments have indeed been addressed.

i0ntempest commented:
> > Are those failing checks just timeouts? If so I'll merge in 24 hours if no one objects.
> 
> You haven't commented or reviewed anything on this PR; so I'd suggest you leave merging of this to the people who did and can verify whether the comments have indeed been addressed.

Cool, I won't touch this then. Just wanted to speed things up a bit since as the author said it's been a month ¯\_(ツ)_/¯

reneeotten commented:
thanks @nilason for all your work on this! I revisited the PR and it seems all good to me, so I'll merge now and if issues show up we can address them!

PS if you're interested in helping to maintain and update these ports, you could consider adding yourself as co-maintainer as @Veence has been rather absent lately.

nilason commented:
> thanks @nilason for all your work on this! I revisited the PR and it seems all good to me, so I'll merge now and if issues show up we can address them!
> 
Thank you all for getting this/these through!

> PS if you're interested in helping to maintain and update these ports, you could consider adding yourself as co-maintainer as @Veence has been rather absent lately.

I've been thinking on this some time, unburden Veence in busy periods. What is the procedure for this?

reneeotten commented:
you can open a PR that adds yourself as a co-maintainer, if @Veence agrees with that (or doesn't respond in a reasonable amount of time) we can just merge it and move on. A slightly more drastic measure would be to file a ["Port abandoned" ticket](https://guide.macports.org/chunked/project.update-policies.html#project.update-policies.abandonment).
